ngraph/core/src/op/einsum.cpp

```diff
@@ -0,0 +1,299 @@
+        NODE_VALIDATION_CHECK(this,
+                              input_type_0 == input_type_i,
+                              "Inputs to Einsum operation must have the same type.");
+    }
+
+    // check that equation has correct format and extract input and output subscripts
+    std::vector<std::string> input_subscripts;
+    std::string output_subscript;
+    parse_equation(m_equation, input_subscripts, output_subscript);
+
+    // a number of input subscripts must match with a number of input tensors
+    NODE_VALIDATION_CHECK(
+        this,
+        input_subscripts.size() == num_inputs,
+        "Equation must contain a number of subscripts equal to a number of Einsum inputs.");
+
+    // create a dictionary with dimension sizes (or ranges in case dynamic shapes) for each label
+    // and check their compatibility in case repeating labels
+    unordered_map<string, PartialShape> label_to_shape;
+    label_to_shape.clear();
+
+    for (size_t input_idx = 0; input_idx < num_inputs; ++input_idx)
+    {
+        const auto& pshape = get_input_partial_shape(input_idx);
+        std::vector<std::string> labels;
+        labels = extract_labels(input_subscripts[input_idx]);
+
+        if (pshape.rank().is_static())
+        {
+            size_t input_rank = pshape.rank().get_length();
+            // check that a rank is greater or equal to a number of labels
+            // these numbers are always equal if there is no ellipsis in the subscript
+            NODE_VALIDATION_CHECK(
+                this,
+                input_rank >= labels.size(),
+                "Input rank must be greater or equal to a number of labels in the "
+                "corresponding input subscript.");
+
+            for (size_t label_ind = 0, dim_ind = 0;
+                 label_ind < labels.size() && dim_ind < input_rank;
+                 ++label_ind)
+            {
+                auto const& label = labels[label_ind];
+                if (label.compare("...") == 0)
+                {
+                    size_t num_broadcasted_dims = input_rank - labels.size() + 1;
+                    auto current_sub_pshape = PartialShape(std::vector<Dimension>(
+                        pshape.begin() + dim_ind, pshape.begin() + dim_ind + num_broadcasted_dims));
+                    if (label_to_shape.find(label) == label_to_shape.end())
+                    {
+                        label_to_shape[label] = current_sub_pshape;
+                    }
+                    else
+                    {
+                        bool is_broadcast_success =
+                            PartialShape::broadcast_merge_into(label_to_shape[label],
+                                                               current_sub_pshape,
+                                                               op::AutoBroadcastType::NUMPY);
+                        NODE_VALIDATION_CHECK(this,
+                                              is_broadcast_success == true,
```

Comment:
remove `== true`

Reply:
I prefer an explicit form because it can be a case that `is_broadcast_success` can change type.

Reply:
Three developers ask you to change this. Please, change.
Follow the "Disagree and commit" approach.

Reply:
Agree so I am done with it. Thanks!